engine/classes/Elgg/Cache/EntityCache.php

```diff
@@ -65,6 +72,20 @@ public function get($guid) {
 	}
 
 	/**
+	 * Returns cached user entity by username
+	 * 
+	 * @param string $username Username
+	 * @return \ElggUser|false
+	 */
+	public function getByUsername($username) {
+		$cache = array_flip($this->username_cache);
```

Comment:
Added in 1.8.0.1 it seems https://github.com/Elgg/Elgg/commit/c80f32c614e48e237bcc3b72b780badf17d99dca